aten/src/ATen/native/ConvolutionMM3d.cpp

```diff
@@ -297,7 +344,25 @@ void slow_conv3d_backward_out_cpu_template(
   fgrad_input.resize_as_(finput);
   TORCH_CHECK(fgrad_input.is_contiguous(), "fgrad_input must be contiguous")
   fgrad_input.zero_();
-  const Tensor tweight2d = weight2d.transpose(0, 1);
+
+  // if the weight is grouped, we need to transpose for each individual
+  // group instead of the entire weight2d
+  Tensor tweight2d;
+  if (groups > 1) {
+    // TODO: find a more elegant way of group transpose
+    tweight2d = at::empty({0}, weight2d.options());
+    tweight2d.resize_as_(weight2d);
+    for (int g = 0; g < groups; ++g) {
+      tweight2d.reshape(
+          {groups, weight2d.size(1), weight2d.size(0) / groups})[g] =
+          weight2d
+              .reshape({groups, weight2d.size(0) / groups, weight2d.size(1)})[g]
+              .transpose(0, 1);
```

Comment:
Not sure if .transpose(0, 1).contiguous() is needed.

Reply:
Looks like it's not needed, 
```
a=torch.randn(2,3,4)
b=torch.arange(24).reshape(2,4,3) # to make sure that it is copied in the correct order. 
a[0]=b[0].transpose(0,1)
```
works and produces expected results. slice of b is copied here, thus it can be discontiguous.

Reply:
I see, we can allow the weights to be discontiguous, right? I will update the impl.

Reply:
What you have is fine, you don't have to change it, and .contiguous() is not necessary